test/general/test_ids.py

```diff
@@ -59,10 +59,24 @@ def test_duplicate_item_ids(self):
         """Test that a game doesn't have item id overlap within its own datapackage"""
         for gamename, world_type in AutoWorldRegister.world_types.items():
             with self.subTest(game=gamename):
-                self.assertEqual(len(world_type.item_id_to_name), len(world_type.item_name_to_id))
+                overlap = False
+                if len(world_type.item_id_to_name) != len(world_type.item_name_to_id):
+                    overlap = \
+                        set(world_type.item_id_to_name.values()) \
+                            .symmetric_difference(set(world_type.item_name_to_id.keys())) or \
+                        set(world_type.item_id_to_name.keys()) \
+                            .symmetric_difference(set(world_type.item_name_to_id.values()))
+                self.assertFalse(overlap)
```

Comment:
Not a big fan of this style of creating boolean values and asserting them, when the real check is actually the if. What was wrong with the suggested code above?